torch/testing/_internal/common_methods_invocations.py

```diff
@@ -400,6 +400,67 @@ def __init__(self,
         #   outside a function's domain.
         self._domain_eps = 1e-5
 
+
+def sample_inputs_binary(op_info, device, dtype, requires_grad):
+    low, high = op_info.domain
+    low = low if low is None else low + op_info._domain_eps
+    high = high if high is None else high - op_info._domain_eps
+
+    test_cases = (
+        # scalar x scalar, scalar x 3d, flippped
+        ((), ()), ((), (S, S, S)), ((S, S, S), ()),
+        # 3d x 2d, 2d x3d, 3d x 3d
+        ((S, S), (S, S, S)), ((S, S, S), (S, S)), ((S, S, S), (S, S, S)),
+        # 3d x 2d broadcasting
+        ((S, 1, S), (M, S)),
+    )
+    return list(
+        SampleInput(make_tensor(s1, device, dtype,
+                                low=low, high=high,
+                                requires_grad=requires_grad),
+                    args=(make_tensor(s2, device, dtype,
+                                      low=low, high=high,
+                                      requires_grad=requires_grad),))
+        for s1, s2 in test_cases)
+
+
+# Metadata class for binary "universal functions (ufuncs)" that accept two
+# tensor and have common properties
+class BinaryUfuncInfo(OpInfo):
+    """Operator information for 'universal binary functions (binary ufuncs).'
+    These are functions of two tensors with common properties like:
+      - they are elementwise functions
+      - the output shape is determined by the input shape
+      - they typically have method and inplace variants
+      - they typically support the out kwarg
+      - they typically have NumPy or SciPy references
+    See NumPy's universal function documentation
+    (https://numpy.org/doc/stable/reference/ufuncs.html) for more details
+    about the concept of ufuncs.
+    """
+
+    def __init__(self,
+                 name,  # the string name of the function
+                 *,
+                 ref,  # a reference function
+                 domain=(None, None),  # the [low, high) domain of the function
+                 dtypes=all_types_and_complex_and(torch.half, torch.bfloat16),
+                 sample_inputs_func=sample_inputs_binary,
+                 **kwargs):
+        super(BinaryUfuncInfo, self).__init__(name,
+                                              dtypes=dtypes,
+                                              sample_inputs_func=sample_inputs_func,
+                                              supports_inplace_autograd=False,
```

Comment:
What happens with this approach if the constructor is used with `supports_inplace_autograd=True`? Won't the OpInfo constructor see two of the same argument?